lib/mempool/mempool.js

```diff
@@ -844,7 +844,20 @@ class Mempool extends EventEmitter {
     const entry = MempoolEntry.fromTX(tx, view, height);
 
     // Contextual verification.
-    await this.verify(entry, view);
+    const conflicts = await this.verify(entry, view);
+
+    // Remove conflicting TXs and their descendants
+    if (conflicts.length) {
+      assert(this.options.replaceByFee);
+
```

Comment:
Its not immediately obvious that `this.verify` will return an empty list if `this.options.replaceByFee` is turned off. Now you need to know implementation specific details of `this.verify` in this context to be sure that the `assert` will never trigger. The same code could be expressed in what feels like a safer way:

```js
if (this.options.replaceByFee) {
    for (const conflict of conflicts) {
      this.logger.debug(
          'Replacing tx %h with %h',
          conflict.tx.hash(),
          tx.hash());
      this.evictEntry(conflict);
    }
}
```

Reply:
Gotcha, this is good feedback thanks. I think we should consider actually just removing `replaceByFee` as an option altogether, like Bitcoin Core.

Also, the options-checks aren't entirely necessary except for the first one here (not affected by this PR):

https://github.com/bcoin-org/bcoin/blob/cec3c3e788834d3e9fc3c0b21bc3e340d573dcfd/lib/mempool/mempool.js#L784-L790

After this point, since all opt-in-RBF TXs are rejected, there will never be any `conflicts` because they would be rejected as double spends. So, the `assert` statement here could be removed. All the other `options.replaceByFee` checks are just optimizations, skipping tests that won't apply if there aren't any RBF TXs in the mempool anyway.

Reply:
I'd vote in favor of leaving it, optimizing for more optionality in the ecosystem. Unless we decide that it would cause a great amount of implementation complexity, which I don't think will.

Reply:
Ok I believe this is more clear now: https://github.com/bcoin-org/bcoin/pull/811/commits/3d5ff9b842497bbf9699d83715f5ccc121fe8ea9